### ISSUE TITLE
Opening an older version ORM file causes crash.

### ISSUE THREAD

**Issue by jpwijbenga** [ormsolutions/NORMA#23]
The solution and orm file were built in VS 2015 with NORMA version 2015-09 CTP from Sourceforge. No projects only one .orm file as a solution item.

Loading the attached files into a bare install of VS2019 CE should lead to reproducing the problem. I selected none of the features, just the bare core, with just the NORMA extension installed on top of this, 

[norma-issue.zip](https://github.com/ormsolutions/NORMA/files/3207256/norma-issue.zip)
Screenshot of exception message:
![afbeelding](https://user-images.githubusercontent.com/6123882/58173715-5d165d80-7c9c-11e9-9129-1b782b5437ff.png)

Likely I missed some (years of) breaking changes? Is there a migration script?

**Comment by mcurland** [comment]
I'm getting a clean open on this file (5 fact types, 5 object types). I'd like to know more of your callstack though. Are you able to create a new .orm file, or is this causing problems as well? A couple of things to try:

1) What happens if you just hit Ignore? Does the file continue to open? Can you see the issue?
2) (Trying to verify the installation here). If you open up another Visual Studio instance and attach a debugger to 'devenv.exe'
  a) Go to the Modules Window (Debug/Windows/Modules)
  b) Sort by Name
  c) Scroll down to find all of the ORMSolutions modules
  d) Select the ORMSolutions rows and copy. This will give you all of the file information from the installation.

**Comment by jpwijbenga** [comment]
1. If I keep hitting ignore (around 10 times it reports): 
![afbeelding](https://user-images.githubusercontent.com/6123882/58235070-5c370780-7d40-11e9-8b48-e2dac3a9d07e.png)
2. Here's the result from the modules window:
[debug.txt](https://github.com/ormsolutions/NORMA/files/3210910/debug.txt)

**Comment by mcurland** [comment]
The files look good, I've tried multiple builds of NORMA here. I guess we can try a full file modules to compare. I'm on 16.0.2, which I isn't latest anymore. What are you on?

You didn't tell me if you can create a new .orm file or not. We can also try different versions of the file with progressively fewer extensions (view off, relational off, abstraction off).

Short of that I'm going to have to put you in a debugger on the source. Debugging is much easier that it used be, but still has a few steps. Bottom line is that short of seeing what property (or properties) are causing problems I'm kind of stuck. There's something out of whack here between whack here between the VS installation and the NORMA files, I just can't tell what it is without a debugger.

**Comment by jpwijbenga** [comment]
I fixed it by Modifying my Visual Studio installation by installing the following group of components.
![afbeelding](https://user-images.githubusercontent.com/6123882/58244650-539bfc80-7d53-11e9-9209-667ecd3ce9c6.png)

I recall that I had previously (before installing VS 2019 however) removed some Visual C runtimes or some Windows SDK features, I thought were not necessary. Maybe installing the .NET desktop environment reinstalled these libraries?

**Comment by jpwijbenga** [comment]
Now trying to uninstall these features and see if I still get the same errors.

**Comment by jpwijbenga** [comment]
Yes, It seems when I uninstall .NET desktop environment components, the errors are back. Is it a known prerequisite? The extension installer did not tell me so. :)

**Comment by jpwijbenga** [comment]
Ok. I didn't answer your questions yet, so here goes (I recreated the failing installation state):

_Are you able to create a new .orm file, or is this causing problems as well?_
Creating a new file works.
_I'm on 16.0.2, which I isn't latest anymore. What are you on?_
I have version 16.1.0
_We can try different versions of the file_
The file with all root elements other than <orm:ORMModel> removed works.
With ormDiagram:ORMDiagram and oial:model elements added, it fails.

**Comment by jpwijbenga** [comment]
[stacktrace.txt](https://github.com/ormsolutions/NORMA/files/3212153/stacktrace.txt)
[Locals.txt](https://github.com/ormsolutions/NORMA/files/3212185/Locals.txt)
Sorry for the formatting (just copy-pasted it). It points to the Absolute bounds property of the objecttypeshape, it seems.

**Comment by mcurland** [comment]
This is really odd. I took off Entity Framework 6 with no difference. The one thing I likely have different is that the Visual Studio Extension workload is selected (with Modeling SDK added) for building NORMA. I haven't tried without this (NORMA needs it to build, so I would have to unininstall/reinstall between builds).
This file is pulled from the installer (the 'More..' button has an 'Export'). Let's compare to yours.

I wouldn't think the desktop files would matter, but they just might. We need either that or a web development I believe to enable C#/VB .NET Framework (old style) projects, which are the only ones that support the NORMA generator right now.

[Matt.vsconfig.txt](https://github.com/ormsolutions/NORMA/files/3213814/Matt.vsconfig.txt)

I'm curious if you can reproduce this with extension development turned on. There are a few files in the NORMA package that are copied from those files, which are necessary for DSL but don't install by default. Maybe I missed one of those, or need another required component in the dependencies list. In any case, when you make changes, export your installer settings. You can then import later to restore.

AbsoluteBounds maps to a RectangleD property type, which is defined in the ...Modeling.Diagrams... assembly, which definitely installs. I just don't see anything obvious here.

My Skype is what it has always been if we need to go there.
